sotodlib/coords/det_match.py

```diff
@@ -167,6 +179,54 @@ def as_array(self):
             data.append(tuple(getattr(r, name) for name, _ in dtype))
         return np.array(data, dtype=dtype)
 
+    @classmethod
+    def from_ctx(cls, ctx, obs_id, stream_id, cal_name='det_cal', am=None):
+        """
+        Load a resonator set from a Context object based on an obs_id
+
+        Args
+        ----------
+        ctx: Context
+            Context object
+        obs_id: str
+            Obs id to base ResSet off
+        stream_id: str
+            Stream id for ResSet to load
+        cal_name: str
+            Attribute of the axismanager that holds detector calibration info
```

Comment:
[As per this](https://github.com/simonsobs/sotodlib/issues/79), a more "sotodlib" way to do this is to have an argument that is the same name as the default field within am, and default it to None.  I.e. `det_cal=None`.  The None will trigger the default to be looked up in `am`; if it isn't None then `det_cal` will be used in place of am[det_cal].  An optional extension to this behavior is to also detect when det_cal is a string and use det_cal = am[det_cal] in that case.

I won't insist on this change, here... depends on how much you see it getting used, how flexible you want to be.